vector/src/main/java/im/vector/app/features/location/MapTilerMapView.kt

```diff
@@ -162,7 +163,7 @@ class MapTilerMapView @JvmOverloads constructor(
         pinDrawable?.let { drawable ->
             if (!safeMapRefs.style.isFullyLoaded ||
                     safeMapRefs.style.getImage(state.pinId) == null) {
-                safeMapRefs.style.addImage(state.pinId, drawable)
+                safeMapRefs.style.addImage(state.pinId, drawable.toBitmap())
```

Comment:
do we know why this behaviour is different? 

`addImage(drawable)` chains into `BitmapUtils.getBitmapFromDrawable(drawable)`  which in turn calls `Bitmap.createBitmap(drawable.getIntrinsicWidth(), drawable.getIntrinsicHeight(), Bitmap.Config.ARGB_8888)`

whereas `drawable.toBitmap()` calls `Bitmap.createBitmap(width, height, config ?: Config.ARGB_8888)`

I would have thought them to be equivalent? :thinking:

Reply:
I wasn't curious enough to understand... Let me look at it to answer this question.

Reply:
After some investigation it seems the only difference I see is that in `BitmapUtils.getBitmapFromDrawable()` they are creating a new drawable with `Drawable drawable = constantState.newDrawable().mutate();` and then the bitmap is created from this new drawable. I suspect the Text part disappears due to this call. In the doc, it says density dependent properties may not be scaled well (https://developer.android.com/reference/android/graphics/drawable/Drawable.ConstantState#newDrawable()). Testing with following code leads to the same issue:
```
val bitmap = drawable.constantState?.newDrawable()?.toBitmap()
bitmap?.let { safeMapRefs.style.addImage(state.pinId, it) }
```

Reply:
great to know, thanks for looking into it! :100: